src/layers.jl

```diff
@@ -127,18 +189,32 @@ end
 @functor Patching
 
 """
-    mlpblock(planes, expansion_factor = 4, dropout = 0., dense = Dense)
+    PosEmbedding{T}
 
-Feedforward block used in many vision transformer-like models.
+Positional embedding layer used by many vision transformer-like models. Instantiated with an 
+embedding vector which is a learnable parameter.
+"""
+struct PosEmbedding{T}
+  embedding_vector::T
+end
+
+(p::PosEmbedding)(x) = x .+ p.embedding_vector[:, 1:size(x)[2], :]
+
+@functor PosEmbedding
 
-# Arguments
-  `planes`: Number of dimensions in the input and output.
-  `hidden_planes`: Number of dimensions in the intermediate layer.
-  `dropout`: Dropout rate.
-  `dense`: Type of dense layer to use in the feedforward block.
-  `activation`: Activation function to use.
 """
-function mlpblock(planes, hidden_planes, dropout = 0., dense = Dense; activation = gelu)
-  Chain(dense(planes, hidden_planes, activation), Dropout(dropout),
-        dense(hidden_planes, planes, activation), Dropout(dropout))
+    CLSTokens{T}
+
+Appends class tokens to the input that are used for classfication by many vision 
+transformer-like models. Instantiated with a class token vector which is a learnable parameter.
+"""
+struct CLSTokens{T}
+  cls_token::T
```

Comment:
Again this isn't a vector. Although it could be, I think. 

Also, I think "CLS" is just "Class"? Not worth saving two letters:
```
struct ClassTokens{T}
    array::T
```

Reply:
Or just `token` for the field. I guess it is repetitive, but `array` is pretty vague IMO.

Reply:
Perhaps we can be even more explicit here with `ClassTokenEmbedding`.